src/sentry/issues/auto_source_code_config/in_app_stack_trace_rules.py

```diff
@@ -0,0 +1,63 @@
+import logging
+from collections.abc import Mapping, Sequence
+from typing import Any
+
+from sentry.issues.auto_source_code_config.code_mapping import CodeMapping
+from sentry.issues.auto_source_code_config.utils import PlatformConfig
+from sentry.models.project import Project
+from sentry.utils import metrics
+
+from .constants import METRIC_PREFIX
+
+logger = logging.getLogger(__name__)
+
+OPTION_KEY = "sentry:automatic_grouping_enhancements"
+
+
+def save_in_app_stack_trace_rules(
+    project: Project, code_mappings: Sequence[CodeMapping], platform_config: PlatformConfig
+) -> list[str]:
+    """Save in app stack trace rules for a given project"""
+    tags: Mapping[str, str | bool] = {
+        "platform": platform_config.platform,
+        "dry_run": platform_config.is_dry_run_platform(),
+    }
+    in_app_stack_trace_rules = generate_in_app_stack_trace_rules(code_mappings)
+    if not platform_config.is_dry_run_platform():
+        project.update_option(OPTION_KEY, "\n".join(in_app_stack_trace_rules))
+
+    metrics.incr(
+        key=f"{METRIC_PREFIX}.in_app_stack_trace_rules.created",
+        amount=len(in_app_stack_trace_rules),
+        tags=tags,
+        sample_rate=1.0,
+    )
+    return in_app_stack_trace_rules
+
+
+def generate_in_app_stack_trace_rules(code_mappings: Sequence[CodeMapping]) -> list[Any]:
+    """Generate in app stack trace rules for a given project"""
```

Comment:
Nit:

```suggestion
    """Generate in-app stack trace rules based on a project's code mappings"""
```

My pretend-I'm-new-to-this brain reads "for a given project" and then gets confused because no project is passed to the function.